assets/js/admin-settings/index.jsx

```diff
@@ -0,0 +1,128 @@
+  const { fields } = wpLocalizedData;
+
+  /**
+   * Helper function for pushing to in-memory settings inside the useSiteOptions hook.
+   */
+   const sendSettings = (updatedRules) => {
+    const next = { ...settings, apple_news_automation: updatedRules };
+
+    // Enforce some defaults prior to save.
+    // Request will 500 when site_logo === null.
+    next.site_logo = next.site_logo ?? 0;
+
+    // Trigger state hook from useSiteOptions.
+    setSettings(next);
+  };
+
+  const addRule = () => {
+    const updatedRules = [...(ruleList ?? [])];
+    updatedRules.unshift({
+      field: '',
+      taxonomy: '',
+      term_id: 0,
+      value: '',
+    });
+    sendSettings(updatedRules);
+  }
+
+  const deleteRule = (ruleIndex) => {
+    const updatedRules = [...(ruleList ?? [])];
+    updatedRules.splice(ruleIndex, 1);
+    sendSettings(updatedRules);
+  }
+
+  /**
+   * Drag and drop logic/re-indexing for Rules.
+   */
+  const reorderRule = () => {
+    // Do nothing if the rule it dropped into its own slot.
+    if (originIndex === targetIndex) {
+      return;
+    }
+    const updatedRules = [...(ruleList ?? [])];
+    // Destructures and reassigns indexed values, effectively swapping them.
+    [updatedRules[originIndex], updatedRules[targetIndex]] = [updatedRules[targetIndex], updatedRules[originIndex]]
+    // Reset draggable indexes.
+    setOriginIndex(null);
+    setTargetIndex(null);
+    sendSettings(updatedRules);
+  }
+
+  /**
+   * Manages Rule component's form state.
+   */
+  const updateRule = (index, key, value) => {
+    const updatedRules = [...(ruleList ?? [])];
+    updatedRules[index] = {...updatedRules[index], [key]: value};
+    // Need to reset value state in case field changes the resulting value's type.
+    if (key === 'field') {
+      updatedRules[index] = {
+        ...updatedRules[index],
+        value: fields[value]?.type === 'boolean' ? 'false' : ''
+      };
+    }
+    sendSettings(updatedRules);
+  }
+
+  return (
+    <div className="apple-news-options__wrapper">
+      <h1>Automation Rules</h1>
+      <Button
+        disabled={busy}
+        isPrimary
+        onClick={saveSettings}
+        style={{ marginRight: '10px' }}
+      >
+        {__('Save Settings', 'apple-news')}
+      </Button>
+      <Button
+        disabled={busy}
+        isPrimary
+        onClick={addRule}
+      >
+        {__('Create New Rule', 'apple-news')}
+      </Button>
+      <h2>Edit Existing Rules</h2>
+      <div style={ruleCorral} className="rule-corral">
+        {!loading && ruleList ? (
+          ruleList.map((item, index) => (
+            <Rule
+              busy={busy}
+              key={index}
+              field={item.field}
+              onDelete={deleteRule}
+              onUpdate={updateRule}
+              reorderRule={reorderRule}
+              ruleIndex={index}
+              ruleList={ruleList}
+              setOriginIndex={setOriginIndex}
+              setTargetIndex={setTargetIndex}
```

Comment:
These could likewise be updated to use the index in this scope vs. passing it into the component:
```suggestion
              onDragStart={() => setOriginIndex(index)}
              onDragOver={(e) => {
                e.preventDefault();
                setTargetIndex(index)
              }}
```

Might also consider putting the logic for `onDragEnd` here too, and then just mapping all three functions in the component back to what is passed in via props. This makes the overall system more flexible - the Rule component won't need to care about what its index is or how items are reordered in the list.